### PR TITLE
Implement full arithmetic flags (NZCV)

### PR DESCRIPTION
**Pull request by HadrienRenaud**
Main contribs:
 - [x] Rename `NZP` into `NZCV`
 - [x] Change flag setting after `ADDS`/`SUBS` and test it
 - [x] Implement full arithmetic condition codes and test it
 - [x] Add a few tests with `BICS` and `ANDS`

Notes:

 - I have included commit 6a70227 that use this implementation for a correct usage in
   ASL. We might want to exclude it, but without it, tests don't pass.

 - I have included a bunch of tests. I think this might be representative
   enough, but I can add more if something comes up. What I have currently is:

    - For flag setting, see tests [`A190`](https://github.com/HadrienRenaud/herdtools7/tree/NZP-NZCV/herd/tests/instructions/AArch64/A190.litmus) to [`A219`](https://github.com/HadrienRenaud/herdtools7/tree/NZP-NZCV/herd/tests/instructions/AArch64/A219.litmus), and [this comment](#issuecomment-1348359229) for a more complete description:
      - `ADDS` and `SUBS`
      - 32 and 64 bits (i.e. operations works on `Xi` or `Wi` registers)
      - 6 cases with sign changes, overflows and carries.

    - For flag setting, see tests `A202` to `A204`, I have a test with `BICS` and 2 with `ANDS`.

    - For condition interpreting, I have three tests where the flags are set,
      and I test all possible branching that will NOT take branch. See for
      example [`A186`](https://github.com/HadrienRenaud/herdtools7/tree/NZP-NZCV/herd/tests/instructions/AArch64/A186.litmus).

### CODE REVIEW

**Comment by HadrienRenaud** [comment]
[Main workflow](https://github.com/HadrienRenaud/herdtools7/actions/workflows/build-versions.yml)

**Comment by HadrienRenaud** [comment]
Pass tests (cf [ci](https://github.com/HadrienRenaud/herdtools7/actions/runs/3678628438))

**Comment by HadrienRenaud** [comment]
A brief summary of added tests for `ADDS` and `SUBS`:

| Test | Op   | Size | Op1                   | Op2                | N | Z | C | V |
| ---- | ---- | ---- | --------------------- | ------------------ | - | - | - | - |
| 190  | ADDS | 64   | \-1                   | 0                  | 1 | 0 | 0 | 0 |
| 191  | ADDS | 64   | 2                     | \-2                | 0 | 1 | 1 | 0 |
| 192  | ADDS | 64   | \-2                   | 5                  | 0 | 0 | 1 | 0 |
| 193  | ADDS | 64   | 2                     | 0x7FFFFFFFFFFFFFFF | 1 | 0 | 0 | 1 |
| 194  | ADDS | 64   |  \-0x8000000000000000 | \-3                | 0 | 0 | 1 | 1 |
| 195  | ADDS | 64   | 2                     | 3                  | 0 | 0 | 0 | 0 |
| 196  | ADDS | 32   | \-1                   | 0                  | 1 | 0 | 0 | 0 |
| 197  | ADDS | 32   | 2                     | \-2                | 0 | 1 | 1 | 0 |
| 198  | ADDS | 32   | \-2                   | 5                  | 0 | 0 | 1 | 0 |
| 199  | ADDS | 32   | 2                     | 0x7FFFFFFF         | 1 | 0 | 0 | 1 |
| 200  | ADDS | 32   |  \-0x80000000         | \-3                | 0 | 0 | 1 | 1 |
| 201  | ADDS | 32   | 2                     | 3                  | 0 | 0 | 0 | 0 |
| 202  |      |      |                       |                    |   |   |   |   |
| 203  |      |      |                       |                    |   |   |   |   |
| 204  |      |      |                       |                    |   |   |   |   |
| 205  | SUBS | 64   | 2                     | 3                  | 1 | 0 | 0 | 0 |
| 206  | SUBS | 64   | \-1                   | 0                  | 1 | 0 | 1 | 0 |
| 207  | SUBS | 64   | 2                     | 2                  | 0 | 1 | 1 | 0 |
| 208  | SUBS | 64   | 2                     | 5                  | 1 | 0 | 0 | 0 |
| 209  | SUBS | 64   | \-2                   | 0x7FFFFFFFFFFFFFFF | 0 | 0 | 1 | 1 |
| 210  | SUBS | 64   |  \-0x8000000000000000 | 3                  | 0 | 0 | 1 | 1 |
| 211  | SUBS | 32   | 2                     | 3                  | 1 | 0 | 0 | 0 |
| 212  | SUBS | 32   | \-1                   | 0                  | 1 | 0 | 1 | 0 |
| 213  | SUBS | 32   | 2                     | 2                  | 0 | 1 | 1 | 0 |
| 214  | SUBS | 32   | 2                     | 5                  | 1 | 0 | 0 | 0 |
| 215  | SUBS | 32   | 0x7FFFFFFF            | \-2                | 1 | 0 | 0 | 0 |
| 216  | SUBS | 32   |  \-0x80000000         | 3                  | 0 | 0 | 1 | 1 |

File: herd/tests/instructions/AArch64/A204.litmus [hunk 1]
```diff
@@ -0,0 +1,21 @@
+AArch64 A203
```
Comment (maranget) [inline herd/tests/instructions/AArch64/A204.litmus:1]: Typo
```suggestion
AArch64 A204
```

File: herd/tests/instructions/AArch64/A204.litmus.expected [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Test A203 Required
```
Comment (maranget) [inline herd/tests/instructions/AArch64/A204.litmus.expected:1]: Consequence of typo
```suggestion
Test A204 Required
```

File: herd/tests/instructions/AArch64/A204.litmus.expected [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+States 1
+0:X3=0; 0:X4=0; 0:X5=0; 0:X6=0;
+No
+Witnesses
+Positive: 0 Negative: 1
+Condition forall (0:X3=1 /\ 0:X4=0 /\ 0:X5=0 /\ 0:X6=0)
+Observation A203 Never 0 1
```
Comment (maranget) [inline herd/tests/instructions/AArch64/A204.litmus.expected:8]:
```suggestion
Observation A204 Never 0 1
```
Additionnaly is it normal for this test to yield `Never` and not `Always`?

**Comment by maranget** [comment]
Thanks a lot @HadrienRenaud, for this very useful work.

**Pull request merged**